### PR TITLE
Drop support for deprecated pack version normalization, display warning about config.yaml deprecation and removal in a future release

### PR DESCRIPTION
**Pull request by Kami**
This pull request includes the following changes:

1. Remove support for pack version normalization (e.g. 2.0 -> 2.0.0). This feature has been deprecated two or three releases ago and schedule for removal in `v2.2.0` (which we haven't done then), so it's time to remove it now.

2. When registering content, if pack contains `config.yaml` file it displays a warning about it being deprecated and replaced with config schemaas and this functionality being removed in `v2.4.0`).

As I mentioned on Slack already, I would be totally fine with removing support for config.yaml in v2.3.0. This feature has been originally deprecated in v1.6.0 which has been released almost a year ago and we have people plenty of warning in the upgrade notes, blog post and documentation.

@dzimine and I were for removal in v2.3.0, but @emedvedev @armab and @LindsayHill were for removal in v2.4.0.

## TODO

- [x] Verify our own CI / CD packs, etc. outside exchange have correct versions

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -97,6 +97,9 @@ in development
 * Add new ``examples.forloop_chain`` action-chain workflow to the examples pack which demonstrates
   how to iterate over multiple pages inside a workflow. #3328
   [Carles Figuerola]
+* Drop support for non-valid semver version strings (e.g. ``2.0``) in pack.yaml pack metadata
+  file. Now only full semver version strings are support (e.g. ``2.0.0``). This functionality has
+  been deprecated for a while and scheduled to be removed in v2.2.0.
```
Comment (LindsayHill) [inline CHANGELOG.rst]: I would re-word this slightly. I would just say:
> Drop support for invalid semver versions strings (e.g. ``2.0``) in ``pack.yaml`` pack metadata.
> Only full semver version strings are supported, e.g. ``2.1.1``. This was originally deprecated in
> v2.1.0.

File: st2common/st2common/bootstrap/base.py [hunk 2]
```diff
@@ -139,6 +139,18 @@ def _register_pack(self, pack_name, pack_dir):
...
+        # Display a warning if pack contains deprecated config.yaml file. Support for those files
+        # will be fully removed in v2.4.0.
+        config_path = os.path.join(pack_dir, 'config.yaml')
+        if os.path.isfile(config_path):
+            LOG.warning('Pack "%s" contains a deprecated config.yaml file (%s).'
+                        'Support for "config.yaml" files has been deprecated in StackStorm v1.6.0 '
+                        'in favor of config.schema.yaml config schema files and config files in '
+                        ' /opt/stackstorm/configs/ directory.'
+                        'Support for config.yaml files will be removed in next major release '
+                        ' (v2.4.0) so you are strongly encouraged to migrate.' %
```
Comment (LindsayHill) [inline st2common/st2common/bootstrap/base.py]: Might be worth adding a link to https://docs.stackstorm.com/reference/pack_configs.html here?

File: CHANGELOG.rst (hunk 1 above)
Comment (Kami) [inline CHANGELOG.rst]: Re-worded, thanks.

File: st2common/st2common/bootstrap/base.py (hunk 2 above)
Comment (Kami) [inline st2common/st2common/bootstrap/base.py]: Agreed, will add it.

**Comment by Kami** [comment]
Verified our packs, addressed the comments.

Will merge once CI passes.

**Pull request merged**